guides/source/sign_up.md

```diff
@@ -0,0 +1,1652 @@
+
+In `config/routes.rb`, let's add a resource for sign up:
+
+```ruby#3
+resource :session
+resources :passwords, param: :token
+resource :sign_up
+```
+
+We're using a singular resource here because we want a singular route for `/sign_up`.
+
+This route directs requests to `app/controllers/sign_ups_controller.rb` so let's create that now.
+
+```ruby
+class SignUpsController < ApplicationController
+  def show
+    @user = User.new
+  end
+end
+```
+
+We're using the `show` action to create a new `User` instance to display the sign up form.
+
+Let's create the form next at `app/views/sign_ups/show.html.erb`:
+
+```erb
+<h1>Sign Up</h1>
+
+<%= form_with model: @user, url: sign_up_path do |form| %>
+  <% if form.object.errors.any? %>
+    <div>Error: <%= form.object.errors.full_messages.first %></div>
+  <% end %>
+
+  <div>
+    <%= form.label :first_name %>
+    <%= form.text_field :first_name, required: true, autofocus: true, autocomplete: "given-name" %>
+  </div>
+
+  <div>
+    <%= form.label :last_name %>
+    <%= form.text_field :last_name, required: true, autocomplete: "family-name" %>
+  </div>
+
+  <div>
+    <%= form.label :email_address %>
+    <%= form.email_field :email_address, required: true, autocomplete: "email" %>
+  </div>
+
+  <div>
+    <%= form.label :password %>
+    <%= form.password_field :password, required: true, autocomplete: "new-password" %>
+  </div>
+
+  <div>
+    <%= form.label :password_confirmation %>
+    <%= form.password_field :password_confirmation, required: true, autocomplete: "new-password" %>
+  </div>
+
+  <div>
+    <%= form.submit "Sign up" %>
+  </div>
+<% end %>
+```
+
+This form collects the user's name, email, and password. We're using the `autocomplete` attribute to suggest to the browser the content requested for these fields.
+
+You'll also notice we set `url: sign_up_path` in the form alongside `model: @user`. Without the `url:` argument, `form_with` would see we have a `User` and send the form to `/users`. Since we want the form to submit to `/sign_up`, the `url:` argument tells the form exactly where to submit the data.
+
+Back in `app/controllers/sign_ups_controller.rb` we can handle the form submission by adding the `create` action.
+
+```ruby#6-19
+class SignUpsController < ApplicationController
+  def show
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(sign_up_params)
+    if @user.save
+      start_new_session_for(@user)
+      redirect_to root_path
+    else
+      render :show, status: :unprocessable_entity
+    end
+  end
+
+  private
+    def sign_up_params
+      params.expect(user: [ :first_name, :last_name, :email_address, :password, :password_confirmation ])
+    end
+end
+```
+
+The `create` action assign parameters and attempts to save the user to the database. If successful, it logs the user in and redirects to `root_path`, otherwise it re-renders the form with errors.
```

Comment:
```suggestion
The `create` action assigns parameters and attempts to save the user to the database. If successful, it logs the user in and redirects to `root_path`, otherwise it re-renders the form with errors.
```